.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '.gitignore'
+      - 'LICENSE'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - '.gitignore'
+      - 'LICENSE'
+      - 'README.md'
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: macos-12
+    env:
+      DEVELOPER_DIR: "/Applications/Xcode_14.1.app/Contents/Developer"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install swift-format
+        run: brew install swift-format
+
+      - name: Lint Code
+        run: make lint
+
+  tests:
+    name: Tests
+    runs-on: macos-12
+    env:
+      DEVELOPER_DIR: "/Applications/Xcode_14.1.app/Contents/Developer"
+
+    steps:
+      - name: Runner Overview
+        run: system_profiler SPHardwareDataType SPSoftwareDataType SPDeveloperToolsDataType
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Run Tests
+        run: make tests
```

Comment:
This line could be replaced with `@./Scripts/run_tests.sh`, why do we need Makefile and why are we launching it through make?

Reply:
The idea behind Makefile here is to use it as a single entry point to the project automation tasks. As of now we only have two such distinctive tasks, namely `lint` and `tests`, but we will likely have more in the future. By invoking the tests through `make` we are making sure that the CI runs the tests in exactly the same fashion as the developer does locally. With that said, if we ever face a need to e.g. add another script that would run Swift tests (with `swift test` command), then we will only need to add the invocation of that script to the `tests` target in the Makefile and everything should work just fine. With the explicit script invocation, as you mentioned, we would need to update CI workflow separately to achieve the same result. I always prefer to have a single source of truth.

Reply:
The tests are run from Xcode though, isn't that a correct assumption for local development? They don't run the run_tests script either, making the point about local development moot in this case.

Reply:
Well, I guess it depends on the individual developer's workflow and habits. If you open up the library as the Xcode project and regularly run the tests on all the supported platforms as you make changes, then I suppose that you won't need to run `make tests` locally. Yet, I personally wouldn't make such an assumption and I would rather leave an option for the developer to run the automated tests locally the same way they are run on CI before pushing the changes and creating PR. Some developers might want to do this as the final check before creating PR - it all depends.

Reply:
So what I see here:

The reason behind moving the functionality in `Makefile` is to use the Workflow file for declaring and enumerating steps for build. All the details are inside of the `Makefile`. I see there an analogue of what we do with Gradle when invoking its tasks, all the details of which are hidden in `build.gradle` files. Gradle is a managed solution though with a lot of default settings that we don't need to think about (and they work absolutely fine for majority of developers). The closest analogy for gradle is fastlane, which is also managed solution with a lot of default settings that just work fine, but we decided to go away from that for the library target. 

Concluding, using Makefile:
**Procs**:
* Slightly better build time
* No overhead for simple tasks (e.g. running unit-tests)
* Better configurability
**Cons**:
* It's our hand-made replacement, I thing a lot of developers will have to think a while to realise the purpose of that scheme
* Another level of abstraction we should keep in mind and spend some time for maintaining

Reply:
Well, to be honest with you, you could attribute exactly the same cons to the previous solution with fastlane. In other words, it is all about having to maintain a super simple Makefile vs Fastfile + its dependencies. It is also worth mentioning that fastlane dependencies need to be updated from time to time and sometimes the major updates require the whole Ruby toolchain to be updated too. With that said, I don't think that having such a simple Makefile requires more maintenance (it's rather opposite in this particular case).

To further expand on this, one could easily say that the use of fastlane for running unit tests alone is an overkill. The fastlane starts to make more sense when you have some more complex automation tasks to be performed. From the fastlane [README](https://github.com/fastlane/fastlane#readme): "fastlane is a tool for iOS and Android developers to automate tedious tasks like generating screenshots, dealing with provisioning profiles, and releasing your application.". So, as the creators of this tool state, it's been designed to simplify *tedious* automation tasks, and thus the fastlane files were moved to `DemoApp` directory with a plan to use fastlane for automating the distribution of the demo app.

Last but not least, in my humble opinion, there is no need to be afraid of the use of Makefile and be worried that a lot of developers may find it confusing. It is actually pretty common for Swift projects to include a Makefile or use CMake (for more complex, multiplatform projects), less common for apps and more common for libraries/tools, and so I don't think that the developers will find it to be an issue. You can take a look at the following repos as an example: [SwiftLint](https://github.com/realm/SwiftLint), [swift-composable-architecture](https://github.com/pointfreeco/swift-composable-architecture), [CryptoSwift](https://github.com/krzyzanowskim/CryptoSwift).